terraform/region/modules/region/waf.tf

```diff
@@ -0,0 +1,150 @@
+resource "aws_wafv2_web_acl" "main" {
+  name        = "${local.account_name}-${local.region_name}-web-acl"
+  description = "Managed rules"
+  scope       = "REGIONAL"
+
+  default_action {
+    allow {}
+  }
+
+  rule {
+    name     = "AWS-AWSManagedRulesPHPRuleSet"
+    priority = 0
+
+    override_action {
+      none {}
+    }
+
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesPHPRuleSet"
+        vendor_name = "AWS"
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "AWS-AWSManagedRulesPHPRuleSet"
+      sampled_requests_enabled   = true
+    }
+  }
+  rule {
+    name     = "AWS-AWSManagedRulesKnownBadInputsRuleSet"
+    priority = 1
+
+    override_action {
+      none {}
+    }
+
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesKnownBadInputsRuleSet"
+        vendor_name = "AWS"
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "AWS-AWSManagedRulesKnownBadInputsRuleSet"
+      sampled_requests_enabled   = true
+    }
+  }
+
+  visibility_config {
+    cloudwatch_metrics_enabled = true
+    metric_name                = "${local.account_name}-${local.region_name}-web-acl"
+    sampled_requests_enabled   = true
+  }
+}
+
+resource "aws_wafv2_web_acl_logging_configuration" "main" {
+  log_destination_configs = [aws_cloudwatch_log_group.waf_web_acl.arn]
+  resource_arn            = aws_wafv2_web_acl.main.arn
+}
+
+resource "aws_cloudwatch_log_group" "waf_web_acl" {
+  name              = "aws-waf-logs-${local.account_name}-${local.region_name}"
+  retention_in_days = 120
```

Comment:
Is 3 months enough for ACL logs? Or is it too costly to store these?

Reply:
its a good question @Raffers  - our counterparts on other teams have used this time period, i can raise it at our CoP to gather an opinion. don't think there was any reason for that other than it might match other retention periods...